app/src/main/java/ch/epfl/sdp/appart/login/LoginService.java

```diff
@@ -28,69 +30,72 @@
      * We do not indicate if the email was found or not to prevent malicious users to know if someone is registered or not
      *
-     * @param email    the user's email
-     * @param callback what should be done after success or failure
+     * @param email the user's email
+     * @return an empty completable future
      * @throws IllegalArgumentException if one of the arguments is null
      */
-    void resetPasswordWithEmail(String email, LoginCallback callback);
+    CompletableFuture<Void> resetPasswordWithEmail(String email);
 
     /**
      * Creates a user with an email and a password
      *
      * @param email    the user's email
      * @param password the user's password
-     * @param callback what should be done after success or failure
+     * @return a completable future containing the User if the request is successful
      * @throws IllegalArgumentException if one of the arguments is null
      * @throws IllegalStateException    if no current user is set
      */
-    void createUser(String email, String password, LoginCallback callback);
+    CompletableFuture<User> createUser(String email, String password);
 
     /**
      * Updates the email address of a user
      *
-     * @param email    the new email to set
-     * @param callback what is performed on update
+     * @param email the new email to set
+     * @return an empty completable future
      * @throws IllegalArgumentException if one of the arguments is null
      * @throws IllegalStateException    if no current user is set
      */
-    void updateEmailAddress(String email, LoginCallback callback);
+    CompletableFuture<Void> updateEmailAddress(String email);
 
     /**
      * Updates the password of a user
      *
      * @param password the new password to set
-     * @param callback what is performed on password change if it succeeds or on failure
+     * @return an empty completable future
      * @throws IllegalArgumentException if one of the arguments is null
      * @throws IllegalStateException    if no current user is set
      */
-    void updatePassword(String password, LoginCallback callback);
+    CompletableFuture<Void> updatePassword(String password);
 
     /**
      * Verifies an user's email
      *
-     * @param callback what is performed on send success or failure
+     * @return an empty completable future
      * @throws IllegalArgumentException if callback is null
      * @throws IllegalStateException    if no current user is set
      */
-    void sendEmailVerification(LoginCallback callback);
+    CompletableFuture<Void> sendEmailVerification();
 
     /**
      * Deletes a user
      *
-     * @param callback what is performed on deletion success or failure
+     * @return an empty completable future
      * @throws IllegalArgumentException if callback is null
      * @throws IllegalStateException    if no current user is set
      */
-    void deleteUser(LoginCallback callback);
+    CompletableFuture<Void> deleteUser();

```

Comment:
In this cases we could return a Future of bool so that we have a way to know if the action was successful, but it's a thing we can do later in the project.

Reply:
Actually, I think we can know the action is successful with the "exceptionally" of the future.